test/e2e/end_user_test.go

```diff
@@ -163,4 +135,99 @@ var _ = Describe("Openshift on Azure end user e2e tests [EndUser]", func() {
+		// Pull the route ingress from the namespace
+		route, err := c.rc.Routes(c.namespace).Get(tpl, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		// make sure only 1 ingress point is returned
+		Expect(len(route.Status.Ingress)).To(Equal(1))
+		host := route.Status.Ingress[0].Host
+		url := fmt.Sprintf("http://%s", host)
+
+		// Curl the endpoint and search for a string
+		httpc := &http.Client{
+			Timeout: 10 * time.Second,
+		}
+		var regex = regexp.MustCompile(`Page views:\s*(\d+)`)
+		By(fmt.Sprintf("hitting the route 3 times, expecting counter to increment (%v)", time.Now()))
+		var prevCounter, currCounter int
+		for i := 0; i < 3; i++ {
+			resp, err := httpc.Get(url)
+			Expect(err).NotTo(HaveOccurred())
+			defer resp.Body.Close()
+			Expect(resp.StatusCode).Should(Equal(http.StatusOK))
+
+			contents, err := ioutil.ReadAll(resp.Body)
+			Expect(err).NotTo(HaveOccurred())
+			matches := regex.FindStringSubmatch(string(contents))
+			Expect(matches).NotTo(BeNil())
+
+			currCounter, err = strconv.Atoi(matches[1])
+			Expect(err).NotTo(HaveOccurred())
+			Expect(currCounter).Should(BeNumerically(">", prevCounter))
+			prevCounter = currCounter
+			time.Sleep(time.Second)
+		}
+
+		// Find the database deploymentconfig and scale to 0
+		var dcName = "postgresql"
+		for i := range []int{0, 1} {
+			By(fmt.Sprintf("searching for the database deploymentconfig (%v)", time.Now()))
+			dbDeploymentConfig, err := c.ac.DeploymentConfigs(c.namespace).Get(dcName, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(dbDeploymentConfig).NotTo(BeNil())
+			By(fmt.Sprintf("scaling the database deploymentconfig to %d (%v)", i, time.Now()))
+			dbDeploymentConfig.Spec.Replicas = int32(i)
+			updated, err := c.ac.DeploymentConfigs(c.namespace).Update(dbDeploymentConfig)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(updated).NotTo(BeNil())
+			By(fmt.Sprintf("waiting for database deploymentconfig to reflect %d replicas (%v)", i, time.Now()))
+			waitErr := wait.PollImmediate(2*time.Second, 10*time.Minute, func() (bool, error) {
+				dc, err := c.ac.DeploymentConfigs(c.namespace).Get(dcName, metav1.GetOptions{})
+				var i32 = int32(i)
+				switch {
+				case err == nil:
+					return i32 == dc.Status.Replicas &&
+						i32 == dc.Status.ReadyReplicas &&
+						i32 == dc.Status.AvailableReplicas &&
+						i32 == dc.Status.UpdatedReplicas &&
+						dc.Generation == dc.Status.ObservedGeneration, nil
+				default:
+					return false, err
+				}
+			})
+			Expect(waitErr).NotTo(HaveOccurred())
+		}
+
+		// wait for the ingress to return 200 in case healthcheck failed when database got recreated
+		By(fmt.Sprintf("making sure the ingress is healthy (%v)", time.Now()))
+		var rt http.RoundTripper
+		waitErr := waitutil.ForHTTPStatusOk(context.Background(), rt, url)
+		Expect(waitErr).NotTo(HaveOccurred())
+
+		By(fmt.Sprintf("hitting the route again, expecting counter to increment from last=%d (%v)", currCounter, time.Now()))
```

Comment:
would prefer this code not to be repeated.  You could define a closure (nested function) if you wanted and call it pre- and post- db restart.